lib/doorkeeper/oauth/authorization_code_request.rb

```diff
@@ -109,6 +117,12 @@ def custom_token_attributes_with_data
           .slice(*Doorkeeper.config.custom_access_token_attributes)
           .symbolize_keys
       end
+
+      def allow_blank_redirect_uri?
```

Comment:
It may be better to phrase this in the way that the spec reads, so: `require_redirect_uri?` and then check `present?` instead of `blank?`